pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symboltable/ClassScope.java

```diff
@@ -230,7 +230,13 @@ public String getClassName() {
         Map<ClassNameDeclaration, List<NameOccurrence>> classDeclarations = getClassDeclarations();
         if (result.isEmpty() && !classDeclarations.isEmpty()) {
             for (ClassNameDeclaration innerClass : getClassDeclarations().keySet()) {
+                ASTMethodDeclarator md = innerClass.getNode().getFirstDescendantOfType(ASTMethodDeclarator.class);
```

Comment:
This doesn't look correct - only the first method is used and added to the images we search for. Which means, we definitely find the first method always whether it's used or not...

In our case, the name occurrence doesn't reference a field or a method, it references a class - e.g. the inner class and this classnamedeclaration is what we should use. Then the search for the next part - the method in this class - can ontinue starting with this class declaration...

This change as it is right now produces false negatives in other rules, e.g. BeanMembersShouldSerialize and false positives in UnusedPrivateField when inner classes are used...

I'll try to work on this...